uPortal-webapp/src/test/java/org/apereo/portal/redirect/PortletRedirectionControllerTest.java

```diff
@@ -57,6 +60,19 @@ public void testExternalUrl() {
         String expected =
                 "http://somewhere.com/something?action=show&list=v1&list=v2&username=student";
         String actual = controller.getUrlString(url, request, new ArrayList<String>());
-        assertEquals(expected, actual);
+        URI uri1 = new URI(expected);
+        URI uri2 = new URI(actual);
+
+        assertTrue(
+                uri1.getPath().equals(uri2.getPath())
+                        && compareQueryParameters(uri1.getQuery(), uri2.getQuery()));
+    }
+
+    private static boolean compareQueryParameters(String query1, String query2) {
+        String[] params1 = query1.split("&");
+        String[] params2 = query2.split("&");
+
+        return Arrays.asList(params1).containsAll(Arrays.asList(params2))
+                && Arrays.asList(params2).containsAll(Arrays.asList(params1));
```

Comment:
would it make sense to use an order insensitive data structure like `Set`, and use set `.equals` to ensure all of the same values are present?